neuro_data/static_images/configs.py

```diff
@@ -427,3 +427,62 @@ def content(self):
 
         def load_data(self, key, **kwargs):
             return super().load_data(key, balanced=True, **kwargs)
+
+    class AreaLayerReliable(dj.Part, AreaLayerRawMixin):
+        definition = """
+        -> master
+        ---
+        stats_source            : varchar(50)  # normalization source
+        stimulus_type           : varchar(512) # type of stimulus
+        exclude                 : varchar(512) # what inputs to exclude from normalization
+        normalize               : bool         # whether to use a normalize or not
+        -> experiment.Layer
+        -> anatomy.Area
+        p_val_power             : tinyint
+        """
+
+        def describe(self, key):
+            return "Like AreaLayer but only neurons that have significantly different (p-val < {:.0E}) response-oracle correlations to the same stimuli vs different stimuli".format(
+                np.power(10, float(key['p_val_power'])))
+
+        @property
+        def content(self):
+            for p in product(['all'],
+                             ['stimulus.Frame', '~stimulus.Frame'],
+                             ['images,responses'],
+                             [True],
+                             ['L2/3'],
+                             ['V1'],
+                             [-3]):
+                yield dict(zip(self.heading.dependent_attributes, p))
+        
+        def load_data(self, key, tier=None, batch_size=1,
+                      Sampler=None, t_first=False, cuda=False):
+            from .stats import BootstrapOracleTTest
+            assert tier in [None, 'train', 'validation', 'test']
+            datasets, loaders = super().load_data(
+                key, tier=tier, batch_size=batch_size, Sampler=Sampler,
+                cuda=cuda)
+            for rok, dataset in datasets.items():
+                member_key = (StaticMultiDataset.Member() & key &
+                            dict(name=rok)).fetch1(dj.key)
+                all_units, all_pvals = (
+                    BootstrapOracleTTest.UnitPValue & member_key).fetch(
+                        'unit_id', 'unit_p_value')
+                assert len(all_pvals) > 0, \
+                    'You forgot to populate BootstrapOracleTTest for group_id={}'.format(
+                    member_key['group_id'])
+                units_mask = np.isin(all_units, dataset.neurons.unit_ids)
```

Comment:
Why don't you first select only the units from `BootstapOracleTTest` that meet the threshold and then do 
`selection = np.isin(dataset.neurons.unit_ids, all_units)`?